Assets/Scripts/SpikeScript.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/***************************************************************************************************
+ * Requires that the player object to have the tag "Player". Each spike will receive a BoxCollider 
+ * and is set on awake to a specific offset and size. The player OnCollision with a spike will 
+ * always be pushed at a 45 degree angle. Intensity is multiplied by the signed Vector. 
+ * Do note, if intensity > 5, then knockback will be greater than the players actual jump height.
+ * Also, this script doesnt disable the players jumping ability. So the player can jump and move
+ * while being knockedback.
+ **************************************************************************************************/
+public class SpikeScript : MonoBehaviour
+{
+    [SerializeField] private float intensity;
+    [SerializeField] private float spikeDamage;
+    private PolygonCollider2D poly;
+
+    void Awake() {
+        poly = gameObject.AddComponent<PolygonCollider2D>();
+        poly.autoTiling = true;
+        intensity = 5f;
+        spikeDamage = 1f;   //TODO :: spikeDamage will probably be greater than 1
+    }
+    void OnCollisionEnter2D(Collision2D other) {
+
+        if (other.gameObject.CompareTag("Player")) {
+            Vector2 playerPos = other.transform.position;
+            ContactPoint2D contacted = other.GetContact(0);  
+            Vector2 direction =  playerPos - contacted.point;
```

Comment:
I think you should get the direction based on the spike position and the player position instead
check this out. The green is the spike collider, the red is the player collider, and the orange arrow is the direction. The player has a capsule collider so its similar to a circle. With your current method, you can hit the spikes on the right side and get knocked back to the left into more spikes. I think it will be better if you always get knocked back away from the spikes. You can do that my using `playerposition - (vector2)transform.position ` . This way you always get knocked away from the center of the spikes
![image](https://user-images.githubusercontent.com/26811871/111018833-309b6c00-8370-11eb-87cb-ab4e19bdfdd0.png)

Reply:
https://github.com/AztecGameLab/LongTermGame5/pull/164/commits/0af7e228e40dcf68f34cd75225ff420ea4b2f373